backend/src/api.py

```diff
@@ -260,12 +276,24 @@ def contribute_review_container_logs(
     return _store_data_for_providers(feedback_input, ProvidersEnum.container, url)
 
 
-@app.get("/frontend/review/random")
-def frontend_review_random():
-    random_feedback_file = Storator3000.get_random()
-    with open(random_feedback_file) as random_file:
-        content = json.loads(random_file.read())
-        return FeedbackSchema(**content).dict() | {"id": random_feedback_file.name.rstrip(".json")}
+@app.get("/frontend/review/{result_id}")
+def frontend_review_random(result_id):
+    if result_id == "random":
+        feedback_file = Storator3000.get_random()
+    else:
+        feedback_file = Storator3000.get_by_id(result_id)
+
+    if not feedback_file:
+        content = {
+            "error": "Server error: 404",
+            "description": f"No result with ID {result_id}",
+        }
+        return JSONResponse(status_code=404, content=content)
```

Comment:
> we have NoDataFound exception for this

which returns 500 rather than 404... I think that's not correct behaviour since it says "hey we did not find these data". Could you please apply this [fix-no-data-found.patch](https://github.com/user-attachments/files/16786885/fix-no-data-found.txt) to `exceptions.py` to fix it, please?

Reply:
Nice, updated :-)